### PR TITLE
Add `BaseChecksum` and API, as well as test fixtures for status JSON

### PR DESCRIPTION
**Pull request by cgwalters**
Add an API to get the base checksum; the MCO wants this.

While we're here add two test fixtures for the status JSON and use it in new unit tests.

### CODE REVIEW

File: pkg/client/client.go [hunk 1]
```diff
@@ -25,6 +25,7 @@ type Deployment struct {
 	ID                      string   `json:"id"`
 	OSName                  string   `json:"osname"`
 	Serial                  int32    `json:"serial"`
+	BaseChecksum            *string  `json:"base-checksum"`
```
Comment (travier) [inline pkg/client/client.go:28]: Any reason we have `*` here when others don't?

Comment (cgwalters) [reply pkg/client/client.go:28]: This field is not always set.  Go doesn't have `Option<T>` so this is the usual way to represent optional values.

File: pkg/client/test-fixtures/fcos-container-status.json [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            "modules": []
+        }
+    ],
+    "transaction": null,
+    "cached-update": null,
+    "update-driver": {
+        "driver-name": "Zincati",
+        "driver-sd-unit": "zincati.service"
+    }
+}
```
[the remark this answers is not in the document]
Comment (jmarrero) [reply pkg/client/test-fixtures/fcos-container-status.json:82]: I guess this json could be auto generated so ignore my comment..

**Review by jmarrero — APPROVED** [review]
/lgtm

**Pull request merged**